### PR TITLE
view attendance

### PR DESCRIPTION
**Pull request by X-workzDev01**

### CODE REVIEW

File: src/main/java/com/xworkz/dream/repository/AttendanceRepository.java [hunk 1]
```diff
@@ -16,5 +16,7 @@ public interface AttendanceRepository {
 	public boolean saveDetilesWithDataSize(List<Object> list, String attendanceRange) throws IOException;
 	
 	public boolean saveDetilesWithoutSize(List<Object> list, String attendanceRange) throws IOException ;
+	
+	public List<List<Object>> getNamesAndCourseName(String spreadsheetId,String range, String value) throws IOException;
```
Comment (X-workzDev01) [inline src/main/java/com/xworkz/dream/repository/AttendanceRepository.java:20]: remove  all public

File: src/main/java/com/xworkz/dream/repository/AttendanceRepositoryImpl.java [hunk 2]
```diff
@@ -124,6 +124,13 @@ public boolean saveDetilesWithoutSize(List<Object> list, String attendanceRange)
 		return true;
 
 	}
+	
+	@Override
+	public List<List<Object>> getNamesAndCourseName(String spreadsheetId,String range, String value) throws IOException {
+		log.info("Reading names and courseName from sheet for spreadsheetId: {}", spreadsheetId);
```
Comment (X-workzDev01) [inline src/main/java/com/xworkz/dream/repository/AttendanceRepositoryImpl.java:130]: remove throws

File: src/main/java/com/xworkz/dream/service/AttendanceService.java [hunk 3]
```diff
@@ -36,7 +38,14 @@ public void markAndSaveAbsentDetails(List<AbsenteesDto> attendanceDtoList, Strin
 
 	public Boolean markTraineeAttendance(String courseName, Boolean batchAttendanceStatus)
 			throws IOException, IllegalAccessException;
+
+	public List<AttendanceDto> addJoined(String courseName) throws IOException, IllegalAccessException;
+
+	public  ResponseEntity<AttendanceDataDto> attendanceReadData(Integer startingIndex, Integer maxRows,
+			String courseName);
+
+	public List<AttendanceDto> filterData(String searchValue, String courseName) throws IOException;
```
Comment (X-workzDev01) [inline src/main/java/com/xworkz/dream/service/AttendanceService.java:48]: remove public and throws

File: src/main/java/com/xworkz/dream/repository/AttendanceRepositoryImpl.java [hunk 4]
```diff
@@ -70,46 +73,67 @@ public boolean writeAttendance(String spreadsheetId, List<Object> row, String ra
...
 		values.add(rowData);
 
 		ValueRange body = new ValueRange().setValues(values);
-		sheetsService.spreadsheets().values().append( sheetId , attendanceRange, body)
-				.setValueInputOption("USER_ENTERED").execute();
-		log.debug("registering fees repository data list is : {}", body);
-
-		return true;
+		try {
+			sheetsService.spreadsheets().values().append(sheetId, attendanceRange, body).setValueInputOption("USER_ENTERED")
```
Comment (xworkzodc) [inline src/main/java/com/xworkz/dream/repository/AttendanceRepositoryImpl.java:122]: Use constant

**Review by xworkzodc — CHANGES_REQUESTED** [review]

**Pull request merged**